x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -1379,6 +1387,65 @@ LogEntryBuilder withRequestBody(DeleteServiceAccountTokenRequest deleteServiceAc
             return this;
         }
 
+        LogEntryBuilder withRequestBody(ActivateProfileRequest activateProfileRequest) throws IOException {
+            logEntry.with(EVENT_ACTION_FIELD_NAME, "activate_user_profile");
+            XContentBuilder builder = JsonXContent.contentBuilder().humanReadable(true);
+            builder.startObject();
+            Grant grant = activateProfileRequest.getGrant();
+            withGrant(builder, grant);
+            builder.endObject();
+            logEntry.with(PUT_CONFIG_FIELD_NAME, Strings.toString(builder));
+            return this;
+        }
+
+        LogEntryBuilder withRequestBody(UpdateProfileDataRequest updateProfileDataRequest) throws IOException {
+            logEntry.with(EVENT_ACTION_FIELD_NAME, "update_user_profile_data");
```

Comment:
I think in this context we've chosen the verb "put" rather than "update", so it would probably be more appropriate to use `put_user_profile_data` for the event action field.

Reply:
This is a good discussion point. I left it to be `update_user_profile_data` for now. We can always decide to change it in a future PR if necessary.

The profile APIs are a bit different than other security APIs in that it probably will have 3 APIs in the `put` category. So far we have `ActivateProfile` and `UpdateProfileData` APIs. In future, we may have a `UpdateProfile` API for maintainence purpose. I think we can agree that they all should use the verb `put` for auditing. But do their `event.action` fields need to begin with that verb, i.e. `put_xxx`? I am not so sure. I personally think it is OK to not begin with `put` and use a word that is more specific to the actual action (e.g. `activate` or `update`). So the two actually forms a hierarchy, e.g. `put` (more generic) and `update_user_profile_data` (more specific). Also the usage of `put` in security APIs normally means it is an "upsert" operation, while `udpate` indicates it does not create new document.